src/views/List.jsx

```diff
@@ -1,13 +1,49 @@
 import { ListItem } from '../components';
+import { useState, useEffect } from 'react';
 
 export function List({ data }) {
+	const [searchTerm, setSearchTerm] = useState('');
+	const [filteredData, setFilteredData] = useState(data);
+
+	const handleChange = (e) => {
+		setSearchTerm(e.target.value);
+	};
+
+	const reset = (e) => {
+		e.preventDefault();
+		setFilteredData(data);
+		setSearchTerm('');
+	};
+
+	useEffect(() => {
```

Comment:
suggestion: I believe the use of `filteredData` state and `useEffect` is redundant because we can utilize a new variable here rather than a state to filter the data. Whenever the `searchTerm` state it will trigger the side effect, which is the same idea as a `useEffect` that will be triggerd because of the change of `[data, searchTerm]`.

```jsx
	const filteredData = data.filter((item) => {
		const itemName = item.name.toLowerCase();
		return itemName.includes(searchTerm.toLowerCase());
	});
```

![CleanShot 2024-02-22 at 10 35 57](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/22878284/a2e5746a-36d6-4db6-a116-34eb6e76c039)

Reply:
Thanks for pointing this out--much simpler!